src/hooks/api/labels.ts

```diff
@@ -111,9 +118,202 @@ const useGetUitpasLabelsQuery = () =>
+    req,
+    queryClient,
+    ...createGetLabelByIdQueryOptions({ id }),
+  });
+
+// Create label
+type CreateLabelArgs = {
+  headers: Headers;
+  name: string;
+  isVisible: boolean;
+  isPrivate: boolean;
+  parentId?: string;
+};
+
+const createLabel = async ({
+  headers,
+  name,
+  isVisible,
+  isPrivate,
+  parentId,
+}: CreateLabelArgs) => {
+  const raw = (name || '').trim();
+  const body: Record<string, unknown> = {
+    name: raw,
+    visibility: isVisible
+      ? LabelVisibilityTypes.VISIBLE
+      : LabelVisibilityTypes.INVISIBLE,
+    privacy: isPrivate ? LabelPrivacyTypes.PRIVATE : LabelPrivacyTypes.PUBLIC,
+  };
+  if (parentId) body.parentId = parentId;
+  return fetchFromApi({
+    path: '/labels/',
+    options: {
+      method: 'POST',
+      headers,
+      body: JSON.stringify(body),
+    },
+  });
+};
+
+const useCreateLabelMutation = (configuration = {}) =>
+  useAuthenticatedMutation({
+    mutationFn: createLabel,
+    mutationKey: 'labels-create',
+    ...configuration,
+  });
+
+// Update visibility/privacy via commands
+type UpdateLabelFlagsArgs = {
+  headers: Headers;
+  id: string;
+  command: 'MakeVisible' | 'MakeInvisible' | 'MakePrivate' | 'MakePublic';
+};
+
+const updateLabel = async ({ headers, id, command }: UpdateLabelFlagsArgs) =>
+  fetchFromApi({
+    path: `/labels/${id}`,
+    options: {
+      method: 'PATCH',
+      headers,
+      body: JSON.stringify({ command }),
+    },
+  });
+
+const useUpdateLabelVisibilityMutation = (configuration = {}) =>
+  useAuthenticatedMutation({
+    mutationFn: ({
+      headers,
+      id,
+      makeVisible,
+    }: {
+      headers: Headers;
+      id: string;
+      makeVisible: boolean;
+    }) =>
+      updateLabel({
+        headers,
+        id,
+        command: makeVisible ? 'MakeVisible' : 'MakeInvisible',
+      }),
+    mutationKey: 'labels-update-visibility',
+    ...configuration,
+  });
+
+const useUpdateLabelPrivacyMutation = (configuration = {}) =>
+  useAuthenticatedMutation({
+    mutationFn: ({
+      headers,
+      id,
+      makePrivate,
+    }: {
+      headers: Headers;
+      id: string;
+      makePrivate: boolean;
+    }) =>
+      updateLabel({
+        headers,
+        id,
+        command: makePrivate ? 'MakePrivate' : 'MakePublic',
+      }),
+    mutationKey: 'labels-update-privacy',
+    ...configuration,
+  });
+
+// Delete label
+const deleteLabel = async ({ headers, id }: { headers: Headers; id: string }) =>
+  fetchFromApi({
+    path: `/labels/${id}`,
+    options: { method: 'DELETE', headers },
+  });
+
+const useDeleteLabelMutation = (configuration = {}) =>
+  useAuthenticatedMutation({
+    mutationFn: deleteLabel,
+    mutationKey: 'labels-delete',
+    ...configuration,
+  });
+
+// Utility: check if a label name is unique (case-insensitive)
+const useIsLabelNameUnique = ({
+  name,
+  currentName,
+}: {
+  name: string;
+  currentName?: string;
+}) => {
+  const raw = (name || '').trim();
```

Comment:
Use a more descriptive name or do the trim directly in the body. 
See previous comment